Render forms on empty product pages [MAILPOET-5859]

johnolek opened the pull request:
## Description

The current logic for form rendering assumes that a single product page will trigger the filter `the_content`. This is not always true. In cases where the product has no description, WooCommerce does not render the description tab at all, which is where that filter usually fires (in my testing).

This change ensures that we still give these forms a chance to render on such pages.

## Code review notes

_N/A_

## QA notes

This does not make the "Below pages" placement option work for product pages without content because they don't render any content. This seems reasonable to me.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5859](https://mailpoet.atlassian.net/browse/MAILPOET-5859)

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[MAILPOET-5859]: https://mailpoet.atlassian.net/browse/MAILPOET-5859?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Commit:
Show forms on empty product pages

The current logic for form rendering assumes that a single product page
will trigger the filter `the_content`. This is not always true. In cases
where the product has no description, WooCommerce does not render the
description tab at all, which is where that filter usually fires (in my
testing).

This change ensures that we still give these forms a chance to render on
 such pages.

MAILPOET-5859

brezocordero reviewed (dismissed):
Nice fix @johnolek 
I have checked if there could be a way to have `woocommerce_product_loop_end` display the form that would also be displayed with `wp_footer` with the new condition and I couldn't find any.
I also tested the form is displayed when viewing a product without a description using Storefront.

Aschepikov commented:
Hi @johnolek there are some failed tests can you rebase it pls

johnolek commented:
@Aschepikov Rebased and looks good to go

Aschepikov commented:
Hi @johnolek  I can't trigger the form on the product with an empty description. I am using all types of forms with different settings and couldn't trigger it on it. Maybe I should do something else or can you check it, I can provide you a creds from my site

johnolek commented:
@Aschepikov It's still working as expected for me in my local environment, so sure, I can take a look at your test site.

Aschepikov commented:
@johnolek Sent you a creds via slack and also check pls the editing of the forms with fixed bar and slide in as I can't edit them and there is a console error
![image](https://github.com/mailpoet/mailpoet/assets/113185437/8d0966f4-de97-453b-961b-f4b0728696ca)
![image](https://github.com/mailpoet/mailpoet/assets/113185437/3dce5a7e-094f-4e00-922e-6a14861121ce)

Commit:
Change method for detecting empty product pages

The previous method was relying on the idea that product pages would not
 trigger the `the_content` filter since they don't have to render the
 content. Testing revealed that this idea was unreliable.

In one instance WordPress was triggering `the_content` when attempting
to filter the excerpt, even though the excerpt was empty.

This changes the detection method to a much more direct inspection of
the post's content.

MAILPOET-5859

Aschepikov reviewed (approved):
GJ!
